### DISCUSSION TITLE
基本类型默认有下面4个能力吗

### DISCUSSION THREAD

**Discussion by liuyangc3** [NonceGeek/Move-Camp#153] [Q&A] [answered]
copy: Allows values of types with this ability to be copied.
drop: Allows values of types with this ability to be popped/dropped.
store: Allows values of types with this ability to exist inside a struct in global storage.
key: Allows the type to serve as a key for global storage operations.

**Comment by Demian101** [comment, marked as answer]
Move 类型的几种能力
 - Copy 被修饰的值可以被复制。
 - Drop 被修饰的值在作用域结束时可以被丢弃。
 - Store 被修饰的值可以被存储到全局状态。
 - Key 被修饰的值可以作为键值对全局状态进行访问。（对应另外一个 value 即 store ， key-store 是一对）


用 key,store 修饰，则表示它不能被复制，也不能被丢弃或重新使用，但是它却可以被安全地存储和转移。
```rust
struct Counter has key, store {  
	value:u64,  
}
```

**整型、布尔等原生类型天然具有 copy、drop、store 能力。**
